Storage/SQL/JoinedHistoryVisitsTable.swift

```diff
@@ -37,7 +37,7 @@ class JoinedHistoryVisitsTable: Table {
         if (cursor.count != 1) {
             return nil
         }
-        return (cursor[0] as Site).id
+        return (cursor[0] as? Site)?.id
```

Comment:
I'm surprised you need optionals here (and in the Favicons one). i.e. we've checked that the cursor has a result. When would it not be a Site?